### PR TITLE
Update video play code in balloon_stress

### PR DESCRIPTION
**Pull request by suqinhuang**
video play is created as a separate case,
run it background in balloon stress test

Signed-off-by: Suqin Huang <shuang@redhat.com>

ID:1130458

### CODE REVIEW

**Comment by suqinhuang** [comment]
https://github.com/autotest/tp-qemu/pull/869

File: qemu/tests/cfg/balloon_stress.cfg [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
-        play_video_cmd  = taskkill /IM wmplayer.exe /F & "${program_files}\Windows Media Player\wmplayer.exe"  "${video_url}"  /play /fullscreen
-        check_playing_cmd = "tasklist|findstr /I wmplayer"
-        driver_name = "balloon.sys"
-        stop_player_cmd = "taskkill /IM wmplayer.exe /F"
-        # reset media player play mode
-        post_cmd += "reg add HKCU\Software\Microsoft\MediaPlayer\Preferences /v ModeLoop /t REG_DWORD /d 0 /f"
+    check_playing_cmd = "tasklist|findstr /I mplayer"
+    driver_name = "balloon.sys"
+    #Disable first startup guide for windows media player
+    wmplayer_reg_cmd = "reg add HKLM\SOFTWARE\Policies\Microsoft\WindowsMediaPlayer /v GroupPrivacyAcceptance  /t REG_DWORD /f /d 00000001"
```
Comment (xutian) [inline qemu/tests/cfg/balloon_stress.cfg]: why still need above command, you can add execute above command in your setup/install.exe

File: qemu/tests/cfg/balloon_stress.cfg [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
+    wmplayer_path = "${program_files}\Windows Media Player\wmplayer.exe"
+    #Install kmplayer if wmplayer is not installed default
+    kmplayer_install_cmd = "start /wait WIN_UTILS:\kmplayer\%s\KMPlayer-setup.exe /SP- /VERYSILENT"
+    kmplayer_path = "${program_files}\KMPlayer\kmplayer.exe"
+    #Disable IEHarden for kmplayer
+    kmplayer_reg_cmd = reg add "HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Internet Settings\ZoneMap" /v IEHarden /t REG_DWORD /f /d 00000000 &
```
Comment (xutian) [inline qemu/tests/cfg/balloon_stress.cfg]: same comment

File: qemu/tests/cfg/balloon_stress.cfg [hunk 3]
```diff
@@ -1,24 +1,26 @@
...
+    play_video_cmd = '"%s" ${video_url} /play /fullscreen'
+    guest_alias = "Win2008-sp2-32:2k8\x86,Win2008-sp2-64:2k8\amd64,Win2008-r2-64:2k8\amd64,Win2012-64:2k12\amd64,Win2012-64r2:2k12\amd64"
```
Comment (xutian) [inline qemu/tests/cfg/balloon_stress.cfg]: is this line still useful?

File: qemu/tests/balloon_stress.py [hunk 4]
```diff
@@ -25,35 +25,35 @@ def run(test, params, env):
...
-    play_video_cmd = params["play_video_cmd"]
-    session.sendline(play_video_cmd)
-    # need to wait for wmplayer loading remote video
-    time.sleep(float(params.get("loading_timeout", 60)))
+    video_play = utils_misc.InterruptedThread(
+        utils_test.run_virt_sub_test, (test, params, env),
+        {"sub_type": params.get("sub_test")})
+    video_play.start()
+    # Wait for media player to load video
+    time.sleep(60)
```
Comment (xutian) [inline qemu/tests/balloon_stress.py]: 60s, looks too long, right?

File: qemu/tests/cfg/balloon_stress.cfg (hunk 2 above)
Comment (suqinhuang) [inline qemu/tests/cfg/balloon_stress.cfg]: It works when I tried at first, but it doesn't work after I put in winutils.iso. I think it's ok add the cfg then I didn't build the installer again :)

File: qemu/tests/cfg/balloon_stress.cfg (hunk 3 above)
Comment (suqinhuang) [inline qemu/tests/cfg/balloon_stress.cfg]: I build separate installer for diff system

File: qemu/tests/balloon_stress.py (hunk 4 above)
Comment (suqinhuang) [inline qemu/tests/balloon_stress.py]: need around 30s in my host to install player and wake up video, I think 60s is ok as some machine maybe slow

File: qemu/tests/cfg/balloon_stress.cfg (hunk 1 above)
Comment (suqinhuang) [inline qemu/tests/cfg/balloon_stress.cfg]: wmplayer is installed default

File: qemu/tests/cfg/balloon_stress.cfg (hunk 2 above)
Comment (suqinhuang) [inline qemu/tests/cfg/balloon_stress.cfg]: updated, could you pls review it

File: qemu/tests/balloon_stress.py (hunk 4 above)
Comment (suqinhuang) [inline qemu/tests/balloon_stress.py]: Win2008 need more than 1 min to install kmplayer

Comment (xutian) [inline qemu/tests/balloon_stress.py]: why not set first = 60 in your wait_for directly?

File: qemu/tests/balloon_stress.py [hunk 5]
```diff
@@ -62,8 +62,7 @@ def run(test, params, env):
         env["balloon_test"] = 1
```
Comment (xutian) [inline qemu/tests/balloon_stress.py]: only need to set once!!

File: qemu/tests/balloon_stress.py (hunk 4 above)
Comment (suqinhuang) [inline qemu/tests/balloon_stress.py]: then it will be flushed with checking cmd

File: qemu/tests/balloon_stress.py (hunk 5 above)
Comment (suqinhuang) [inline qemu/tests/balloon_stress.py]: seems it's used in driver_in_use

Comment (xutian) [inline qemu/tests/balloon_stress.py]: I know, only set one time is enough. not need to write again and again!!

File: qemu/tests/balloon_stress.py [hunk 6]
```diff
@@ -25,45 +25,39 @@ def run(test, params, env):
...
+        lambda: utils_misc.get_guest_cmd_status_output(
+            vm, check_playing_cmd)[0] == 0, first=60, timeout=600)
     if not running:
-        raise error.TestError("Video is not playing")
+        raise exceptions.TestError("Video is not playing")
 
-    #for case:balloon_in_use to call
-    env["balloon_test"] = 0
-    error.context("balloon vm memory in loop", logging.info)
+    # for case:balloon_in_use to call
```
Comment (xutian) [inline qemu/tests/balloon_stress.py]: drop it

**Pull request merged**